modules/apps/client-extension/client-extension-web/test/js/AttribureFields.test.tsx

```diff
@@ -0,0 +1,43 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {fireEvent, render} from '@testing-library/react';
+import React from 'react';
+
+import AttributeFields from '../../src/main/resources/META-INF/resources/js/components/global-js/AttributeFields';
+
+const onAttributeChange = jest.fn();
+
+const mockAttributeFields = () => (
+	<AttributeFields index={0} onAttributeChange={onAttributeChange} value="" />
+);
+
+describe('AttributeFields', () => {
+	it('Prevents user from adding normal whitespace on attribute field', () => {
+		const {getByTestId} = render(mockAttributeFields());
+
+		const attributeField = getByTestId('testId_0');
+
+		fireEvent.change(attributeField, {target: {value: 'name with space'}});
+
+		expect(onAttributeChange).toHaveBeenCalledTimes(1);
+		expect(onAttributeChange).toHaveBeenCalledWith(0, {
+			name: 'namewithspace',
+		});
+	});
+
+	it('Prevents user from adding non-breaking-space on attribute field', () => {
+		const {getByTestId} = render(mockAttributeFields());
+
+		const attributeField = getByTestId('testId_0');
+
+		fireEvent.change(attributeField, {target: {value: '\u00A0 \u00A0'}});
```

Comment:
It would be better to test a more real world scenario: `name with non breaking space` where the spaces are non breaking spaces. Otherwise, it could be that just calling trim would cause this test to pass.

Reply:
Make sense, @ethib137. 
Thanks!